ElementX/Sources/Screens/HomeScreen/View/Filters/FilterView.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2024 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct FilterView: View {
+    let filter: RoomListFilter
+    @StateObject var state: RoomListFiltersState
+
+    var body: some View {
+        let binding = Binding<Bool>(get: {
+            state.isSelected(filter)
+        }, set: { isSelected, _ in
+            state.setIsEnabled(isSelected, for: filter)
+        })
+        Toggle(isOn: binding) {
+            Text(filter.localisedName)
+        }
+        .toggleStyle(FilterToggleStyle())
+    }
+}
+
+struct FilterView_Previews: PreviewProvider, TestablePreview {
+    static var previews: some View {
+        FilterView(filter: .people, state: .init())
+        FilterView(filter: .people, state: .init(enabledFilters: [.people]))
+    }
+}
+
+private struct FilterToggleStyle: ToggleStyle {
+    private func strokeColor(isOn: Bool) -> Color {
+        isOn ? .compound.bgActionPrimaryRest : .compound.borderInteractiveSecondary
+    }
+    
+    private func backgroundColor(isOn: Bool) -> Color {
+        isOn ? .compound.bgActionPrimaryRest : .compound.bgCanvasDefault
+    }
+    
+    private func foregroundColor(isOn: Bool) -> Color {
+        isOn ? .compound.textOnSolidPrimary : .compound.textPrimary
+    }
+    
+    func makeBody(configuration: Configuration) -> some View {
+        let shape = RoundedRectangle(cornerRadius: 20)
+        configuration.label
+            .font(.compound.bodyLG)
+            .foregroundColor(foregroundColor(isOn: configuration.isOn))
+            .padding(.horizontal, 16)
+            .padding(.vertical, 8)
+            .background(shape.fill(backgroundColor(isOn: configuration.isOn)))
+            .overlay(shape
+                .inset(by: 0.5)
+                .stroke(strokeColor(isOn: configuration.isOn)))
```

Comment:
Tiny nit: I find using a view builder easier to read than a multiline parameter.
```suggestion
            .overlay {
                shape
                    .inset(by: 0.5)
                    .stroke(strokeColor(isOn: configuration.isOn))
            }
```